### ISSUE TITLE
Players time out seconds after joining

### ISSUE THREAD

**Issue by Budderman18** [GeyserMC/Geyser#3965]
### Describe the bug

All be players using 1.20.10 will time out incredibly fast.

### To Reproduce

join a server with the latest geyser with 1.20.1

### Expected behaviour

Player time out only after a very long time of losing connection

### Server Version and Plugins

This server is running Pufferfish version git-Pufferfish-16 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: af1f56c on HEAD)

Server Plugins (72):
[00:51:43 INFO]: Bukkit Plugins:
[00:51:43 INFO]:  - *AdvancedReplay, *AdvancedSecurity, ajParkour, AntiCheatReplay, *AutoKiller, BadlionClientModAPI, BedWars, BentoBox, BentoBox-Biomes, BentoBox-BSkyBlock
[00:51:43 INFO]:  BentoBox-Challenges, BentoBox-Level, BentoBox-Warps, Citizens, ClearLag, CommandSigns, ConsoleSpamFix, CrazyAuctions, Curses, debuginfo-be
[00:51:43 INFO]:  Duels, Essentials, EssentialsChat, EssentialsDiscord, EssentialsGeoIP, EssentialsWarpGUI, F3NPerm, Factions, floodgate, Geyser-Spigot
[00:51:43 INFO]:  GriefPrevention, GroupManager, IngotBridges, IngotHub, IngotMinigamesAPI, IngotServerManager, IngotSkywars, IngotSpleefPlus, IngotSurvivalGames, IngotVanillaTweaker
[00:51:43 INFO]:  IngotWarn, IngotWorldManager, JukeBox, *KillerMoney, LibsDisguises, *LunarClient-API, *LunarClient-Impl, MobArena, Morphy, Multiverse-Core
[00:51:43 INFO]:  Multiverse-Inventories, Multiverse-Portals, *MultiWorldMoney, NoteBlockAPI, PlaceholderAPI, PlotSquared, Prison, ProtocolLib, RandomTeleport, SignShop
[00:51:43 INFO]:  SkyblockOres, spark, StaffChat, Themis, TimeManager, TitleManager, *TowerDefense, UltraCosmetics, Vault, Votifier
[00:51:43 INFO]:  VotingPlugin, WorldEdit

### Geyser Dump

https://dump.geysermc.org/eioWoT2FhcNFkcUK1GjqMdBzMgaOS2st

### Geyser Version

This server is running Geyser version 2.1.2-SNAPSHOT (git-master-3a0c1b7) (Java: 1.20, Bedrock: 1.19.80/1.19.81 - 1.20.10)

### Minecraft: Bedrock Edition Device/Version

IPhone XR/1.20.10 + Windows11/1.20.10

**Comment by tegelsoep** [comment]
Hi, I got the same problem.

But I got a temporary fix. Disable `forward-player-ping` in the geyser config.

Hope this works!

**Comment by PedroMPagani** [comment]
Can confirm the behavior, on DonutSMP.net. forward player ping seems to be causing it.

**Comment by Impactic48** [comment]
I can confirm as well. Forward player ping seems to be causing this issue.

**Comment by BackSam6348** [comment]
This issue is also happening in Standalone. It seems like a forward-player-ping issue.

**Comment by CallumADW** [comment]
setting forward-player-ping to false also fixed the issue for me.

**Comment by PedroMPagani** [comment]
Have any of you that changed the forward-player-ping to false are getting random disconnects from time to time? of still timeouts.

**Comment by tegelsoep** [comment]
> Have any of you that changed the forward-player-ping to false are getting random disconnects from time to time? of still timeouts.

Nope, after I disabled `forward-player-ping`, I haven't had any more problems at all.

**Comment by XiaMoHuaHuo-CN** [comment]
> 你们中是否有人将前向玩家 ping 更改为 false 时不时地随机断开连接？的静止超时。

Nope

**Comment by BinaryBible** [comment]
My server is also experiencing this issue. Error included.

`15.07 01:09:29 [Server] [ERROR] [org.cloudburstmc.protocol.bedrock.netty.codec.packet.BedrockPacketCodec]: Error encoding packet SetEntityDataPacket(metadata={FLAGS=[CAN_SHOW_NAME, SILENT, CAN_CLIMB, HAS_COLLISION, HAS_GRAVITY, SCENTING], , runtimeEntityId=301, tick=0, properties=EntityProperties(intProperties=[], floatProperties=[]))
15.07 01:09:29 [Server] [INFO] org.cloudburstmc.protocol.bedrock.codec.PacketSerializeException: Error whilst serializing SetEntityDataPacket(metadata={FLAGS=[CAN_SHOW_NAME, SILENT, CAN_CLIMB, HAS_COLLISION, HAS_GRAVITY, SCENTING], , runtimeEntityId=301, tick=0, properties=EntityProperties(intProperties=[], floatProperties=[]))
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.BedrockCodec.tryEncode(BedrockCodec.java:80) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.netty.codec.packet.BedrockPacketCodec.encode(BedrockPacketCodec.java:39) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.netty.codec.packet.BedrockPacketCodec.encode(BedrockPacketCodec.java:19) ~[?:?]
15.07 01:09:29 [Server] [INFO] at io.netty.handler.codec.MessageToMessageCodec$1.encode(MessageToMessageCodec.java:67) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:90) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:879) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:851) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1010) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.AbstractChannel.write(AbstractChannel.java:296) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.BedrockPeer.onTick(BedrockPeer.java:91) ~[?:?]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:159) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:406) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[velocity-3.2.0-SNAPSHOT-260.jar:3.2.0-SNAPSHOT (git-37fedf50-b260)]
15.07 01:09:29 [Server] [INFO] at java.lang.Thread.run(Thread.java:833) [?:?]
15.07 01:09:29 [Server] [INFO] Caused by: java.lang.IllegalArgumentException: Failed to encode EntityData 0 of java.util.EnumSet
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v361.BedrockCodecHelper_v361.writeEntityData(BedrockCodecHelper_v361.java:138) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v291.serializer.SetEntityDataSerializer_v291.serialize(SetEntityDataSerializer_v291.java:19) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v557.serializer.SetEntityDataSerializer_v557.serialize(SetEntityDataSerializer_v557.java:13) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v557.serializer.SetEntityDataSerializer_v557.serialize(SetEntityDataSerializer_v557.java:9) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.BedrockCodec.tryEncode(BedrockCodec.java:78) ~[?:?]
15.07 01:09:29 [Server] [INFO] ... 23 more
15.07 01:09:29 [Server] [INFO] Caused by: java.lang.IllegalArgumentException: No id found for SCENTING
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.common.util.Preconditions.checkArgument(Preconditions.java:207) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.common.util.TypeMap.getId(TypeMap.java:39) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.transformer.FlagTransformer.serialize(FlagTransformer.java:28) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.transformer.FlagTransformer.serialize(FlagTransformer.java:14) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v361.BedrockCodecHelper_v361.writeEntityData(BedrockCodecHelper_v361.java:104) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v291.serializer.SetEntityDataSerializer_v291.serialize(SetEntityDataSerializer_v291.java:19) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v557.serializer.SetEntityDataSerializer_v557.serialize(SetEntityDataSerializer_v557.java:13) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.v557.serializer.SetEntityDataSerializer_v557.serialize(SetEntityDataSerializer_v557.java:9) ~[?:?]
15.07 01:09:29 [Server] [INFO] at org.cloudburstmc.protocol.bedrock.codec.BedrockCodec.tryEncode(BedrockCodec.java:78) ~[?:?]
15.07 01:09:29 [Server] [INFO] ... 23 more`

**Comment by BinaryBible** [comment]
> Have any of you that changed the forward-player-ping to false are getting random disconnects from time to time? of still timeouts.

This just caused me to go from being able to connect and then getting kicked, to being completely unable to connect at all.

**Comment by MrSono77** [comment]
> > Have any of you that changed the forward-player-ping to false are getting random disconnects from time to time? of still timeouts.
> 
> Confirmed. I disable forward-player-ping, and players in my server got randomly kicked through out the day. Sometime 30 players, sometime 50 players. (and they're all bedrock)

What is your server version? Is your server on 1.20?

**Comment by MinerCoffee** [comment]
1.20.1 and yes.

**Comment by PedroMPagani** [comment]
> > Have any of you that changed the forward-player-ping to false are getting random disconnects from time to time? of still timeouts.
> 
> Confirmed. I disable forward-player-ping, and players in my server got randomly kicked through out the day. Sometime 30 players, sometime 50 players. (and they're all bedrock)

So your timeouts are LESS, but still also happen?

**Comment by MrSono77** [comment]
> 1.20.1 and yes.

This is awesome. How can you keep your server stable with so many players, my server overloaded with only 9 players and me using a bunch of optimization mods. I could really use some tips, if it's not too much to ask of course

**Comment by Koekieezz** [comment]
> > 1.20.1 and yes.
> 
> This is awesome. How can you keep your server stable with so many players, my server overloaded with only 9 players and me using a bunch of optimization mods. I could really use some tips, if it's not too much to ask of course

proxies, ig.

**Comment by Konicai** [comment]
Fixed in #3993

**Comment by Synkstar** [comment]
Still is broken for me when forward player ping is on.

**Comment by onebeastchris** [comment]
@Synkstar Please send a geyser dump (run /geyser dump" in the server console, and send the link here), also, which device are you joining on? aka a ps4, mobile, win10?

**Comment by Synkstar** [comment]
https://dump.geysermc.org/EOmQ0glLpgdRDJWzbylLdnlx6GPoo0H8. I am joining from android 1.20.10

**Comment by onebeastchris** [comment]
And does the issue go away if you disable forward-player-ping?
If it doesn't; this is most likely an entirely separate issue.

**Comment by Synkstar** [comment]
It does go away if you disable forward-player-ping

**Comment by Konicai** [comment]
@Synkstar can you join our discord and ping me

**Comment by LimesKey** [comment]
Fixed for me!

**Comment by Synkstar** [comment]
> LimesKey
Do you have forward-player-ping on ?

**Comment by LimesKey** [comment]
@Synkstar Yes I do

**Comment by Impactic48** [comment]
This issue is not fixed. It is still timing players out with build #244... @Konicai 

Geyser Dump https://dump.geysermc.org/TS4T1WNTaUh3QtYbBqjPtQoDX7zsUKZM
